doc/languages-frameworks/go.section.md

```diff
@@ -62,6 +62,82 @@ The following is an example expression using `buildGoModule`:
 }
 ```
 
+### Obtaining and overriding `vendorHash` for `buildGoModule` {#buildGoModule-vendorHash}
+
+`nix-prefetch` can be used to obtain the actual hash. The following command gets the value of `vendorHash` for package `pet`:
+
+```sh
+cd path/to/nixpkgs
+nix-prefetch -E "{ sha256 }: ((import ./. { }).my-package.overrideAttrs (_: { vendorHash = sha256; })).go-modules"
+```
+
+To obtain the hash without external tools, set `vendorHash = lib.fakeSha256;` and run the build. ([more details here](#sec-source-hashes)).
+
+`vendorHash` can be overridden with `overrideAttrs`. Override the above example like this:
+
+```nix
+{
+  pet_0_4_0 = pet.overrideAttrs (
+    finalAttrs: previousAttrs: {
+      version = "0.4.0";
+
+      src = fetchFromGitHub {
+        inherit (previousAttrs.src) owner repo;
+        rev = "v${finalAttrs.version}";
+        hash = "sha256-gVTpzmXekQxGMucDKskGi+e+34nJwwsXwvQTjRO6Gdg=";
+      };
+
+      vendorHash = "sha256-dUvp7FEW09V0xMuhewPGw3TuAic/sD7xyXEYviZ2Ivs=";
+    }
+  );
+}
+```
+
+### Overriding `goModules` (#buildGoModule-goModules-override)
+
+As `goModules` would become `""` when the final state of attribute `vendorHash` becomes `null`, overriding `goModules` directly with `goModules.overrideAttrs` may cause error when subsequent overriding specifies `venhorHash = null`.
+
+The following Nix expression error out `error: value is a string while a set was expected`, because
```

Comment:
Could you write please as a prepending sentence, next to this one, what exactly do the two examples (the not evaluating one and the good one) demonstrate? What does `FOO = "foo";` mean?